File: backends/ruby-hanami/app/domain/notification/repository.rb
module Domain
  module Notification
    module Repository
      def list_page(staff_id, cursor, limit)                                     = raise NotImplementedError
      def counts(staff_id)                                                       = raise NotImplementedError
      def bulk_mark_read(staff_id, ids, all)                                     = raise NotImplementedError
      def store(staff_id, message_type, title, message, created_by, url: nil)   = raise NotImplementedError
      def patch(id, attrs)                                                       = raise NotImplementedError
    end
  end
end

File: backends/go-beego/internal/domain/notification/repository.go
package notification

type Repository interface {
	ListPage(staffID uint, cursor *string, limit int) ([]*Notification, *string, error)
	Counts(staffID uint) (unread, total int64, err error)
	BulkMarkRead(staffID int64, ids []int64, all bool) (int64, error)
	Store(staffID uint, messageType int, title, message string, createdBy uint, url ...string) error
	Patch(id int64, attrs map[string]interface{}) (bool, error)
}
